tianshou/data/batch.py

```diff
@@ -269,14 +269,13 @@ def __init__(self,
                 for k, v in batch_dict.items():
                     if isinstance(v, (list, tuple, np.ndarray)):
                         v_ = None
-                        if isinstance(v, np.ndarray):
-                            v_ = v
-                        elif all(isinstance(e, torch.Tensor) for e in v):
+                        if not isinstance(v, np.ndarray) and \
```

Comment:
`if not isinstance(v, np.ndarray)` part is unnecessary no ? Since `np.array` of `torch.Tensor` is impossible.

Reply:
In this branch, ``v`` can be ``list, tuple, np.ndarray``.  ``if not isinstance(v, np.ndarray)`` means ``isinstance(v, (list, tuple))``, then I can iterate them at a low cost.

Reply:
I don't get it. `all(isinstance(e, torch.Tensor) for e in np.array([1.0, 2.0]))` is fine. It has a higher cost than for list or tuple even using short circuiting ?

Reply:
``for e in np.array([1.0, 2.0])`` should have a higher cost, because numpy has to construct each ``e``.

Reply:
```
In [16]: a = np.zeros((10000000, 10))                                                                                                                                                                       

In [17]: %timeit all(isinstance(e, torch.Tensor) for e in a)                                                                                                                                                
481 ns ± 1.74 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [18]: a2 = a.tolist()                                                                                                                                                                                    

In [19]: %timeit all(isinstance(e, torch.Tensor) for e in a2)                                                                                                                                               
386 ns ± 3.09 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Are you talking about this ?

Reply:
Yes, this is what I mean. You are such a quick coder :)